api/src/main/java/org/apache/cloudstack/api/command/user/volume/MigrateVolumeCmd.java

```diff
@@ -120,4 +120,19 @@ public void execute() {
         }
     }
 
+    @Override
+    public String getSyncObjType() {
+        if (getSyncObjId() != null) {
+            return BaseAsyncCmd.migrationSyncObject;
+        }
+        return null;
+    }
+
+    @Override
+    public Long getSyncObjId() {
+        if (getStoragePoolId() != null) {
```

Comment:
If `getStoragePoolId()` != null, then return `getStoragePoolId()`, otherwise return null....

You can simple call `return getStoragePoolId()`. The result is the same.

Reply:
the consideration here (and the intermediate version) that other objects, like the host, could be the sync object as well so this could be extended to return that under conditions.